### PR TITLE
Creating a custom page in SonataAdmin

### PR DESCRIPTION
**Pull request by kunicmarko20**
I am targeting this branch, because this is BC.

## Subject

Added docs for creating a custom page from my [blog post](https://medium.com/@kunicmarko20/sonata-admin-custom-page-61b99d4bd9d4).

### CODE REVIEW

File: docs/cookbook/recipe_custom_page.rst [hunk 1]
```diff
@@ -0,0 +1,126 @@
+Creating a Custom Admin Page
+============================
+
+Sometimes you may want to create an additional page in your SonataAdmin
+but you do not have or need an Entity to connect it to and you want to
+add it to sidebar with the rest of your admin pages, here we will present
+a full working example of how is that achievable.
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_page.rst]: You should add some punctuation in there IMO 😅

File: docs/cookbook/recipe_custom_page.rst [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+The recipe
+----------
+
+SonataAdmin provides a very straight-forward way of adding your own custom
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_page.rst]: straightforward, no hyphen

File: docs/cookbook/recipe_custom_page.rst [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+- create an admin class for your custom page
+- extend the ``SonataAdmin:CRUD`` Controller and tell our admin class to
+  use it
+- create a template that will represent your custom page
+
+Create an admin class
+^^^^^^^^^^^^^^^^^^^^^
+
+First we need to create an Admin class and we will clear all routes except
+list.
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_page.rst]: Quote this with double backticks?

File: docs/cookbook/recipe_custom_page.rst [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                - null
+                - null
+                - App\Controller\PageCRUDController
+            public: true
+
+.. note::
+    If you are not using Symfony Flex, this service should be registered
+    in ``app/config/services.yml``.
+
+For more information about service configuration please refer to Step 3 of :doc:`../getting_started/creating_an_admin`
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_page.rst]: Should probably be linebroken

**Comment by OskarStark** [comment]
could you add a look and feel screenshot of the blank page including the link in the sidebar to the docs?

maybe "Page" is a bit weird and can confuse users with the PageBundle?

Great article so far 👍

**Comment by kunicmarko20** [comment]
Maybe call it a "static Page"?

**Comment by jordisala1991** [comment]
What about custom view?

**Comment by kunicmarko20** [comment]
Renamed to view but I am not sure I like it, it can easily get confused with e.g. form/list view

**Comment by kunicmarko20** [comment]
Could you review this one more time and what do you think about "custom view" name instead of "custom page"?

File: docs/cookbook/recipe_custom_view.rst [hunk 5]
```diff
@@ -0,0 +1,134 @@
+Creating a Custom Admin View
+============================
+
+Sometimes you may want to create an additional view in your SonataAdmin
+but you do not have or need an Entity to connect it to. You also probably
+want to add it to sidebar with the rest of your admin pages? Here we will
+present a full working example of how is that achievable.
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_view.rst]: - "Sometimes,"
- "Sonata admin"
- "to the sidebar"
- "admin pages."
- "how that is achievable"

File: docs/cookbook/recipe_custom_view.rst [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+SonataAdmin provides a very straightforward way of adding your own custom
+view.
+
+To do this we need to:
+
+- create an admin class for your custom view
+- extend the ``SonataAdmin:CRUD`` Controller and tell our admin class to
+  use it
+- create a template that will represent your custom view
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_view.rst]: Please end bullet points with semi-colon, and the last one with a dot:
- this;
- that;
- and finally, this.

File: docs/cookbook/recipe_custom_view.rst [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+.. note::
+    If you are not using Symfony Flex, the namespace will be `AppBundle`
+    instead of `App`.
+
+Extending the Admin Controller
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Create your own Controller extending the one from SonataAdmin
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_view.rst]: Either end this with a dot or a colon

File: docs/cookbook/recipe_custom_view.rst [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+  use it
+- create a template that will represent your custom view
+
+Create an admin class
+^^^^^^^^^^^^^^^^^^^^^
+
+First we need to create an admin class and we will clear all routes except
+``list``.
+
+.. code-block:: php
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_view.rst]: From https://symfony.com/doc/current/contributing/documentation/standards.html :

> The :: shorthand is preferred over .. code-block:: php to begin a PHP code block (read the Sphinx documentation to see when you should use the shorthand);

File: docs/cookbook/recipe_custom_view.rst [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+
+    {% block sonata_admin_content %}
+        Your content here
+    {% endblock %}
+
+.. note::
+    If you are not using Symfony Flex, this template should be saved to
+    ``app/Resources/views/admin/custom_view.html.twig``
+
+The final view will look like:
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_view.rst]: "like this"

File: docs/cookbook/recipe_custom_view.rst [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+        protected $baseRoutePattern = 'custom_view';
+        protected $baseRouteName = 'custom_view';
+
+        protected function configureRoutes(RouteCollection $collection)
+        {
+            $collection->clearExcept(['list']);
+        }
+    }
+
+.. note::
```
Comment (jordisala1991) [inline docs/cookbook/recipe_custom_view.rst]: I do not know if this note is necessary really, this namespace depends on the application. Im +1 on using how flex comes by default. But if you are not using flex, I guess you know how to change this to adapt to your bundle namings, right? wdyt?

File: docs/cookbook/recipe_custom_view.rst (hunk 8 above)
Comment (kunicmarko20) [reply docs/cookbook/recipe_custom_view.rst]: Did not know this, in docs it is mostly ` .. code-block:: php`

File: docs/cookbook/recipe_custom_view.rst (hunk 10 above)
Comment (kunicmarko20) [reply docs/cookbook/recipe_custom_view.rst]: I agree, removed.

File: docs/cookbook/recipe_custom_view.rst (hunk 8 above)
Comment (greg0ire) [reply docs/cookbook/recipe_custom_view.rst]: Yeah most people don't know about this rule, but it exists :P

File: docs/cookbook/recipe_custom_view.rst [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+    {% block sonata_admin_content %}
+        Your content here
+    {% endblock %}
+
+The final view will look like this:
+
+.. figure:: ../images/custom_admin_view.png
+    :align: center
+    :alt: Custom view
+        :width: 700px
```
Comment (jordisala1991) [inline docs/cookbook/recipe_custom_view.rst]: wrong indent

File: docs/cookbook/recipe_custom_view.rst [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+The recipe
+----------
+
+SonataAdmin provides a very straightforward way of adding your own custom
+view.
+
+To do this we need to:
+
+- create an admin class for your custom view;
```
Comment (core23) [inline docs/cookbook/recipe_custom_view.rst]: Why did you add `;` at the end of the lines?

Comment (kunicmarko20) [reply docs/cookbook/recipe_custom_view.rst]: >greg0ire a day ago  Member
Please end bullet points with semi-colon, and the last one with a dot:
this;
that;
and finally, this.

/cc @greg0ire :smile:

Comment (greg0ire) [reply docs/cookbook/recipe_custom_view.rst]: Because I asked him, and also that's also how you are supposed to format bullet lists, unless it contains sentences, in which case you should end them with a dot.

Comment (greg0ire) [reply docs/cookbook/recipe_custom_view.rst]: Might be considered old-fashioned though: https://english.stackexchange.com/questions/75591/semicolons-at-the-end-of-list-items#75619

**Comment by jordisala1991** [comment]
Thank you @kunicmarko20
